packages/optimistic-oracle/src/proposer.js

```diff
@@ -191,29 +309,45 @@ class OptimisticOracleProposer {
 
   // Sets allowances for all collateral currencies used in unproposed price requests
   async _setAllowances() {
-    // TODO: Should we do this also for currencies from undisputedProposals?
-    for (let priceRequest of this.optimisticOracleClient.getUnproposedPriceRequests()) {
-      // The OO requires approval to transfer the proposed price request's collateral currency in order to post a bond.
-      // We'll set this once to the max value and top up whenever the bot's allowance drops below MAX_INT / 2.
+    const approvalPromises = [];
+
+    // Increase allowance to MAX for the `priceRequest.currency`
+    const _approveCollateralCurrencyForPriceRequest = async priceRequest => {
       const collateralToken = new this.web3.eth.Contract(getAbi("ExpandedERC20"), priceRequest.currency);
-      const [currentCollateralAllowance] = await Promise.all([
-        collateralToken.methods.allowance(this.account, this.ooContract.options.address).call()
-      ]);
+      const currentCollateralAllowance = await collateralToken.methods
+        .allowance(this.account, this.ooContract.options.address)
+        .call();
       if (this.toBN(currentCollateralAllowance).lt(this.toBN(MAX_UINT_VAL).div(this.toBN("2")))) {
-        const collateralApprovalTx = await collateralToken.methods
+        const collateralApprovalPromise = collateralToken.methods
           .approve(this.ooContract.options.address, MAX_UINT_VAL)
           .send({
             from: this.account,
             gasPrice: this.gasEstimator.getCurrentFastPrice()
+          })
+          .then(tx => {
+            this.logger.info({
+              at: "OptimisticOracle#Proposer",
+              message: "Approved OO to transfer unlimited collateral tokens 💰",
+              currency: collateralToken.options.address,
+              collateralApprovalTx: tx.transactionHash
+            });
           });
-        this.logger.info({
-          at: "OptimisticOracle#Proposer",
-          message: "Approved OO to transfer unlimited collateral tokens 💰",
-          currency: collateralToken.options.address,
-          collateralApprovalTx: collateralApprovalTx.transactionHash
-        });
+        approvalPromises.push(collateralApprovalPromise);
       }
+    };
+
+    // The OO requires approval to transfer the proposed price request's collateral currency in order to post a bond.
+    // We'll set this once to the max value and top up whenever the bot's allowance drops below MAX_INT / 2.
+    for (let priceRequest of this.optimisticOracleClient.getUnproposedPriceRequests()) {
```

Comment:
This is where [this new contract event](https://github.com/UMAprotocol/protocol/pull/2546/files#r577015775) is used.